pkg/chart/chart.go

```diff
@@ -326,28 +327,38 @@ func (t *Testing) InstallChart(chart string, valuesFiles []string) TestResult {
 
 	result := TestResult{Chart: chart}
 
-	if len(valuesFiles) > 0 {
-		for _, valuesFile := range valuesFiles {
-			release, namespace := util.CreateInstallParams(chart, t.config.BuildId)
+	// Test with defaults if no values files are specified.
+	if len(valuesFiles) == 0 {
+		valuesFiles = append(valuesFiles, "")
+	}
+
+	for _, valuesFile := range valuesFiles {
+		var namespace, release, releaseSelector string
 
+		if t.config.Namespace != "" {
+			namespace = t.config.Namespace
+			release, _ = util.CreateInstallParams(chart, t.config.BuildId)
+			releaseSelector = fmt.Sprintf("%s=%s", t.config.ReleaseLabel, release)
+			defer t.helm.DeleteRelease(release)
```

Comment:
`defer t.helm.DeleteRelease(release)` could be moved to after the `if/else` block.